src/spdx_tools/spdx3/bump_from_spdx2/relationship.py

```diff
@@ -1,74 +1,277 @@
+        swap_direction = parameters_for_bump[2]
+    if swap_direction:
+        if not to:
+            print_missing_conversion("Swapped Relationship to NoAssertion/ None", 0)
+            return
+        from_element = to[0]
+        to = [spdx2_relationship.spdx_element_id]
+    else:
+        from_element = spdx2_relationship.spdx_element_id
+
+    return Relationship(
+        spdx_id,
+        creation_information,
+        from_element,
+        to,
+        relationship_type,
+        comment=spdx2_relationship.comment,
+        completeness=completeness,
     )
 
 
-def bump_relationship_type(spdx2_relationship_type: Spdx2_RelationshipType) -> Optional[Tuple[RelationshipType, bool]]:
-    if spdx2_relationship_type == Spdx2_RelationshipType.DESCRIBED_BY:
-        return RelationshipType.DESCRIBES, True
-    if spdx2_relationship_type == Spdx2_RelationshipType.CONTAINED_BY:
-        return RelationshipType.CONTAINS, True
-    if spdx2_relationship_type == Spdx2_RelationshipType.DEPENDENCY_OF:
-        return RelationshipType.DEPENDS_ON, True
-    if spdx2_relationship_type == Spdx2_RelationshipType.GENERATED_FROM:
-        return RelationshipType.GENERATES, True
-    if spdx2_relationship_type == Spdx2_RelationshipType.HAS_PREREQUISITE:
-        return RelationshipType.PREREQUISITE, True
-    if spdx2_relationship_type.name.endswith("_OF"):
-        relationship_type = spdx2_relationship_type.name.replace("_OF", "")
-        return RelationshipType[relationship_type], False
-    if spdx2_relationship_type.name.endswith("_FOR"):
-        relationship_type = spdx2_relationship_type.name.replace("_FOR", "")
-        return RelationshipType[relationship_type], False
-    return RelationshipType[spdx2_relationship_type.name], False
-    # if spdx2_relationship_type == Spdx2_RelationshipType.PATCH_APPLIED:
-    #     print_missing_conversion("RelationshipType.PATCH_APPLIED", 0)
-    #     return None
+def determine_completeness(
+    related_spdx_element_id: Union[str, SpdxNone, SpdxNoAssertion]
+) -> Tuple[Optional[RelationshipCompleteness], List[str]]:
+    if isinstance(related_spdx_element_id, SpdxNoAssertion):
+        completeness = RelationshipCompleteness.NOASSERTION
+        to = []
+    elif isinstance(related_spdx_element_id, SpdxNone):
+        completeness = RelationshipCompleteness.COMPLETE
+        to = []
+    else:
+        completeness = None
+        to = [related_spdx_element_id]
+    return completeness, to
+
+
+def merge_relationships_and_add_to_payload(payload: Payload, relationships: List[Relationship]):
+    merged_relationship = relationships[0]
+    for relationship in relationships[1:]:
+        merged_relationship.to += relationship.to
+        if merged_relationship.comment and relationship.comment:
+            merged_relationship.comment += ", " + relationship.comment
+        if not merged_relationship.comment and relationship.comment:
+            merged_relationship.comment = relationship.comment
```

Comment:
This overwrites the completeness of all relationship with that of the first one. I believe that if one of them has NOASSERTION, the resulting relationship should have, too.
There might be cases where the completeness statements contradict each other, we have to think how to deal with that.

Reply:
You are right, I also thought about contradicting relationships and I am still not sure if this merigng approach is a good idea. Maybe we should only merge relationships if they contain no additional information at all, so only merge "simple" from-relationship-type-to-spdx-element relationships. And as soon as there is a comment or `SpdxNoAssertion` or `SpdxNone` involved, don't merge anything...